go/server/handlers.go

```diff
@@ -20,21 +20,48 @@ package server
 
 import (
 	"github.com/gin-gonic/gin"
+	"github.com/vitessio/arewefastyet/go/tools/macrobench"
 	"github.com/vitessio/arewefastyet/go/tools/microbench"
 	"log"
 	"net/http"
 	"sort"
 )
 
+func handleRenderErrors(c *gin.Context, err error) {
+	if err == nil {
+		return
+	}
+	log.Println(err.Error())
+	c.HTML(http.StatusOK, "error.tmpl", gin.H{
+		"title":    "Vitess benchmark - Error",
+		"url":      c.FullPath(),
+		"errorStr": err.Error(),
+	})
+}
+
 func (s *Server) informationHandler(c *gin.Context) {
 	c.HTML(http.StatusOK, "information.tmpl", gin.H{
 		"title": "Vitess benchmark",
 	})
 }
 
-func (s *Server) homeHanlder(c *gin.Context) {
+func (s *Server) homeHandler(c *gin.Context) {
+	oltpData, err := macrobench.GetResultsForLastDays("ee", "webhook", 31, s.dbClient)
+	if err != nil {
+		handleRenderErrors(c, err)
+		return
+	}
+
+	tpccData, err := macrobench.GetResultsForLastDays(macrobench.TPCC, "webhook", 31, s.dbClient)
+	if err != nil {
+		handleRenderErrors(c, err)
```

Comment:
we should not return from first error. If we can show the second widget, we should show that atleast

Reply:
Resolved in 6383b39abc7496ef2ce05777b36bd323d14d6a72. Errors are logged serverside, but the page will be rendered. If some graphs can't be rendered, the text  _`The graph could not be rendered.`_ will be displayed instead.